### PR TITLE
Switch to Ruff

### PR DESCRIPTION
**Pull request by tuomas777**
Replaced Black, Flake8 and isort with [Ruff](https://docs.astral.sh/ruff/). Similar to [Linked Events](https://github.com/City-of-Helsinki/linkedevents/pull/1005) and [Kerro Kantasi](https://github.com/City-of-Helsinki/kerrokantasi/pull/522), but a lot less code changes were needed.

Also marked one possible bug to be checked and fixed later https://github.com/City-of-Helsinki/hauki/pull/248/commits/09c4114667596564bb324e3168f55ee5cacb257d.

**NOTE**: A couple of tests fail in Azure, but the same ones fail also in the main branch, so those shouldn't be related to this PRs changes.

Refs [HAUKI-666](https://github.com/City-of-Helsinki/hauki/pull/248/commits/09c4114667596564bb324e3168f55ee5cacb257d) 👿 

[HAUKI-666]: https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

File: requirements-dev.in [hunk 1]
```diff
@@ -15,6 +15,4 @@ requests-mock
 safety
 
 # Keep these tools in sync with .pre-commit-config.yaml
-black==24.4.2
-flake8==7.1.0
-isort==5.13.2
+ruff
```
Comment (danipran) [inline requirements-dev.in:18]:
```suggestion
ruff==0.7.0
```

**Review by danipran — APPROVED** [review]
LGTM! Just need to pin ruff version

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HAUKI-API branch is deployed to platta: https://hauki-pr248.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## api tests result is __success__ for https://hauki-pr248.api.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**